### PR TITLE
don't return an error if the singleton rbac doesn't exist

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**
Doesn't return an error from `ensureSingletonRBAC` if we can detect that the cache is just a bit behind

**Motivation for the change:**
Fixes https://github.com/operator-framework/operator-lifecycle-manager/issues/2091

prior to this change, I could reproduce with:
```
$ go test ./pkg/controller/operators/olm/... -v -run ^TestSyncOperatorGroups$ -count 10
```
after this change, I couldn't reproduce with:
```
$ go test ./pkg/controller/operators/olm/... -v -run ^TestSyncOperatorGroups$ -count 100
```

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/controller/operators/olm/operatorgroup.go
Comment (timflannagan) [inline pkg/controller/operators/olm/operatorgroup.go:495]: nit:

```suggestion
				if k8serrors.IsAlreadyExists(err) && ownerutil.IsOwnedByLabel(cr, csv) {
					continue
				}
				return err
```

**Comment by timflannagan** [comment]
Nice :sunglasses: 
Holding for another opinion (if needed) but this looks reasonable.

/lgtm
/hold

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Comment by dinhxuanvu** [comment]
This PR failed one e2e test `creation with nodeSelector config`.

**Review by njhale — APPROVED** [review]
I think it's also time to resurrect the flake analyzer

**Pull request merged**
